### PR TITLE
Docs: update GitLab OAuth2 documentation

### PR DESCRIPTION
**Pull request by IevaVasiljeva**
**What is this feature?**

Updating the GitLab OAuth2 documentation:

* restructuring it to follow the layout set https://github.com/grafana/grafana-authnz-team/issues/223;
* adding a table of config options;
* correcting some outdated documentation (role mapping behaviour);
* lots of rewording etc.

Also setting the default GitLab scopes to `openid email profile` instead of `api`. `api` scope gives access that is unnecessarily wide, and `openid email profile` have been tested to work fine for a new GitLab integration. It is also consistent with the docs that we had.

**Why do we need this feature?**

To make our docs easier to use.

**Who is this feature for?**

Users of Grafana's public docs

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana-authnz-team/issues/230

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md [hunk 1]
```diff
@@ -19,85 +19,83 @@ weight: 1000
...
+| `api_url`                    | Yes      | Grafana uses `<api_url>/user` endpoint to obtain GitLab user information compatible with [OpenID UserInfo](https://connect2id.com/products/server/docs/api/userinfo).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | `https://gitlab.com/api/v4`          |
```
Comment (Jguer) [inline docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:81]:
```suggestion
| `api_url`                    | No      | Grafana uses `<api_url>/user` endpoint to obtain GitLab user information compatible with [OpenID UserInfo](https://connect2id.com/products/server/docs/api/userinfo).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | `https://gitlab.com/api/v4`          |
```

Should not be necessary with openid scope

File: docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md [hunk 2]
```diff
@@ -19,85 +19,83 @@ weight: 1000
...
 
-`tls_skip_verify_insecure` controls whether a client verifies the server's certificate chain and host name. If it is true, then SSL/TLS accepts any certificate presented by the server and any host name in that certificate. _You should only use this for testing_, because this mode leaves SSL/TLS susceptible to man-in-the-middle attacks.
+| Setting                      | Required | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | Default                              |
```
Comment (Jguer) [inline docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:72]: nit/not needed for merge/seeking opinion: Is the order we present these options the best?

Should it be required first? 
Isn't enabled required for pratical purposes :D?

**Review by Jguer — APPROVED** [review]

File: docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md [hunk 3]
```diff
@@ -19,85 +19,83 @@ weight: 1000
...
+   Review the list of other GitLab [configuration options]({{< relref "#configuration-options" >}}) and complete them, as necessary.
+
+1. Optional: [Configure a refresh token]({{< relref "#configure-a-refresh-token" >}}):
 
-You may have to set the `root_url` option of `[server]` for the callback URL to be
-correct. For example in case you are serving Grafana behind a proxy.
+   a. Enable `accessTokenExpirationCheck` feature toggle.
 
-Restart the Grafana backend for your changes to take effect.
+   c. Set `use_refresh_token` to `true` in `[auth.generic_oauth]` section in Grafana configuration file.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:60]: what happened to `b`?

**Review by chri2547 — APPROVED** [review]
Just one question from me. Looking great!

File: docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md (hunk 2 above)
Comment (IevaVasiljeva) [reply docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:72]: Good point, good point :) I'll reorder them to have the required options first, and also set `enabled` to required.

File: docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md (hunk 3 above)
Comment (IevaVasiljeva) [reply docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md]: Hehe, well spotted! Got swallowed by `c` 🙈

**Comment by morsik** [comment]
Looks like this PR removed information about great `is_admin` feature that I found here: https://github.com/grafana/grafana/issues/28892

Is this feature removed? Because I can't make `is_admin && "GrafanaAdmin"` to work...

**Comment by morsik** [comment]
@IevaVasiljeva this seems weird... thread you pointed to is 7 years old, but use of `is_admin` flag in Grafana was added 4 years ago.

Also... your example in docs about possibility of using just simple e-mail to mark user as `GrafanaAdmin` also doesn't work, so... I'm out of options how to provide myself full Grafana Admin access when using Gitlab auth :/

**Comment by IevaVasiljeva** [comment]
@morsik nothing forbids you from using `is_admin` in your Grafana SSO configuration as long as your GitLab version supports it. This PR just updated the docs so that Grafana docs wouldn't be misleading for people who are on newer GitLab versions that don't support `is_admin` anymore.

**Comment by morsik** [comment]
> newer GitLab versions that don't support `is_admin` anymore

@IevaVasiljeva could you please explain more about this? I'm almost always on latest version on Gitlab, so that might be my problem. Since when this is not supported anymore? ;(

**Comment by IevaVasiljeva** [comment]
@morsik I've been looking at the code some more, and I was wrong about why `is_admin` field is not returned anymore. It's true that it's been removed from some GitLab endpoints, but not from the one that we use (`/api/v4/user`, [GitLab docs](https://docs.gitlab.com/ee/api/users.html#as-an-administrator-2)). The actual reason why `is_admin` is not returned anymore is a backend change as you initially suspected. We switched over to using OIDC compliant endpoints ([PR](https://github.com/grafana/grafana/pull/69890)), which do not return the `is_admin` field.

To give you a bit more context on how Grafana fetches user info from GitLab:
* first we try to extract it from the ID token using [GitLab's `oauth/token` and `oauth/userinfo` endpoints](https://docs.gitlab.com/ee/api/oauth2.html#cross-origin-resource-sharing). These endpoints comply with OIDC standard and do not provide `is_admin` field (related [Grafana code](https://github.com/grafana/grafana/blob/v11.4.0/pkg/login/social/connectors/gitlab_oauth.go#L271-L309));
* if no ID token is provided or we can't extract the info, we try to fetch user info from GitLab's `/api/v4/user` endpoint ([related Grafana code](https://github.com/grafana/grafana/blob/v11.4.0/pkg/login/social/connectors/gitlab_oauth.go#L235-L269)).

@Jguer do you know if there is a way to force the usage of `/api/v4/user` to fetch user info?

My apologies for making the wrong assumptions earlier @morsik!

**Comment by morsik** [comment]
@IevaVasiljeva no worries! Just trying to find out why I can't make automatic Admins in my Grafana when Gitlab Admin is logging to it via Gitlab OAuth because none of the examples in documentation worked ;)

**Pull request merged**